### PR TITLE
Sessions view

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes
- implement #882 

## Todo:
- [x] paginated
- [x] surface all related properties to the person in session
~- [ ] add migrations to create materialized table for prexisting data~
- [ ] red indicator for currently in progress

## Checklist
- [x] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)
- [ ] Cypress E2E tests (if applicable)

### CODE REVIEW

**Comment by EDsCODE** [comment]
@timgl Didn't add the red indicator for in progress yet and decided to hold off on materialized view until followup ticket  because i tried it and it works but there's lengthier testing and config that needs to be done

File: frontend/src/scenes/sessions/SessionsTable.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        },
+        {
+            title: 'Start Time',
+            render: function RenderStartTime(session) {
+                return <span>{humanFriendlyDetailedTime(session.start_time)}</span>
+            },
+        },
+        {
+            title: 'Start Point',
+            render: function RenderStartPoint(session) {
```
Comment (timgl) [inline frontend/src/scenes/sessions/SessionsTable.js:45]: Would be good to add ellipsis: true to start and end point too

**Review by timgl — COMMENTED** [review]
Looks good. I do think we should wait with merging this until we have the materialised view ready too, because it is very slow in current state.

**Comment by EDsCODE** [comment]
@timgl This is ready for another look! I ended up not using materialized views or any database specific optimization strategy because it seemed overkill to manage. I figured out a format of the feature that makes the queries really light. Instead of trying to compute the entire sessions table, I've setup the page to work by day. This actually makes the browsing experience better because you can actually scroll to a specific day and find what people were up to. This way, the queries can be extremely precise and quick. 

However, the one inconvenience happens when you reach the end of a specific day. There is a way to fix this but it would lead to another problem where sparsely logged events would make pagination on this page really weird. LMK what you think!

File: frontend/src/scenes/sessions/SessionDetails.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                return <span>{humanFriendlyDetailedTime(timestamp)}</span>
+            },
+        },
+    ]
+
+    return (
+        <Table
+            columns={columns}
+            rowKey={event => event.id}
+            dataSource={events}
```
Comment (timgl) [inline frontend/src/scenes/sessions/SessionDetails.js:38]: Suggest adding `pagination={{ pageSize: 50, hideOnSinglePage: true }}` or something here, it's a bit annoying having to navigate through

File: frontend/src/scenes/sessions/SessionDetails.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                if (!event) return { props: { colSpan: 0 } }
+                let param = event.properties['$current_url'] ? '$current_url' : '$screen_name'
+                return <Property value={event.properties[param]} />
+            },
+            ellipsis: true,
+        },
+        {
+            title: 'Time',
+            render: function RenderEvent({ timestamp }) {
+                return <span>{humanFriendlyDetailedTime(timestamp)}</span>
```
Comment (timgl) [inline frontend/src/scenes/sessions/SessionDetails.js:29]: I'm not sure this is the right format for this table (I do think it is for the other one). What I'd want most I think is:
- timestamp including seconds
- difference in time between the previous event and this one

File: frontend/src/scenes/sessions/SessionsTable.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                return (
+                    <span>
+                        {!_.isEmpty(session.events) && _.first(session.events).properties?.$current_url
```
Comment (timgl) [inline frontend/src/scenes/sessions/SessionsTable.js:50]: NIT: Wonder if you could hide the `https://` from this url too

**Review by timgl — COMMENTED** [review]
When scrolling through this, I really found myself wanting to filter this just by people with email addresses. Could be another ticket though!

Couple small things below

File: frontend/src/scenes/sessions/SessionDetails.js (hunk 2 above)
Comment (EDsCODE) [reply frontend/src/scenes/sessions/SessionDetails.js:50]: done!

**Comment by timgl** [comment]
Mergy conflicts

**Comment by mariusandra** [comment]
I found one thing: when you change the pagination, the "order" still starts from "1", even if it should probably start from 51:

image

image

image

**Pull request merged**